File: 1291-sequential-digits/1291-sequential-digits.cpp
class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        vector<int> ans;
        queue<int> q;
        for (int i = 1; i <= 9; ++i) {
            q.push(i);
        }

        while (!q.empty()) {
            int num = q.front();
            q.pop();
            if (num > high) {
                return ans;
            }
            if (num >= low && num <= high) {
                ans.push_back(num);
            }
            int lastDigit = num % 10;
            if (lastDigit < 9) {
                q.push(num * 10 + lastDigit + 1);
            }
        }

        return ans;
    }
};

File: 1291-sequential-digits/1291-sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        ans = []
        q = deque([1, 2, 3, 4, 5, 6, 7, 8, 9])

        while q:
            num = q.popleft()
            if num > high:
                return ans
            if low <= num <= high:
                ans.append(num)
            lastDigit = num % 10
            if lastDigit < 9:
                q.append(num * 10 + lastDigit + 1)

        return ans